### PR TITLE
use RecipeRule in clusterUI rule pushing

### PR DESCRIPTION
**Pull request by barentine**
Addresses issue was running into issues with multiline str yaml recipe text in `HTTPRulePusher` with json dumps/loads required to replace our templated USERFILEs, and I am more invested in refactored rules working, and encountered less json headaches this way anyway.

see #476 

**Is this a bugfix or an enhancement?**
bugfix
**Proposed changes:**
- remove task pusher check
- user RecipeRule to push itself rather than HTTPRulePusher

**Checklist:**

- [ ] Tested with numpy=1.14
1.16
- [ ] Tested on python 2.7 and 3.6
3.6
- [ ] Tested with wx=3.x and wx=4.x [if UI code]
- [x] Does the PR avoid variable renaming in existing code, whitespace changes, and other forms of tidying? [There is a place for code tidying, but it makes reviewing 
much simpler if this is kept separate from functional changes]

### CODE REVIEW

**Comment by David-Baddeley** [comment]
Could you provide a clear explanation of what the problem you are trying to address is (file an issue?).

The stuff in `cluster._rules` is a temporary stopgap for use with your acquisition recipe chaining, but which is slated for removal (as discussed previously) once we've done some more refactoring. Although some refactoring of rule pushing is on the TODO list, I'm obviously not super keen on replacing code which doesn't depend on something which is going to be removed with code that does.

**Comment by barentine** [comment]
see #487. 

`cluster._rules` may not be the final refactor of rules that push themselves, but looking at what you envision in your #372 comment, I don't see how whatever supersedes it wouldn't be a drop-in in this context.

I of course prefer to work with forward-progress code: `cluster._rules` is closer to an eventual `cluster.rules` than `HTTPRulePusher.RecipePusher`, so I think we should move on with it.

**Comment by barentine** [comment]
@David-Baddeley updated to use `cluster.rules.RecipeRule` rather than `cluster._rules`.

File: PYME/cluster/clusterUI/recipes/views.py [hunk 1]
```diff
@@ -28,31 +29,30 @@ def get_input_glob(request):
...
-
-    fileNames = request.POST.getlist('files', [])
-    pusher.fileTasksForInputs(input=fileNames)
+    from PYME.cluster.rules import RecipeRule
+    import posixpath
 
+    recipeURI = ('pyme-cluster://%s/' % server_filter) + request.POST.get('recipeURL').lstrip('/')
+    input_list = request.POST.getlist('files', [])
+    try:
+        # put outputs in the same directory as selected files, under 'analysis'
```
Comment (David-Baddeley) [inline PYME/cluster/clusterUI/recipes/views.py]: I don't like this as it has the potential to get super messy if you are analysing files across directories. Approach to date was to hard-code the output directory in the recipe / not rely on substitution (this is safe, if gross), a better option might be to have it as a UI input as in `run_template()`.

Comment (barentine) [reply PYME/cluster/clusterUI/recipes/views.py]: how do you analyze files across directories in this view?

Comment (barentine) [reply PYME/cluster/clusterUI/recipes/views.py]: agree ui input would be best

Comment (David-Baddeley) [reply PYME/cluster/clusterUI/recipes/views.py]: This is used by `recipe_standalone()` which accepts globs for inputs.

Comment (David-Baddeley) [reply PYME/cluster/clusterUI/recipes/views.py]: as well as the lightweight form in the files view.

Comment (David-Baddeley) [reply PYME/cluster/clusterUI/recipes/views.py]: Pre-populating the output directory as current directory + /analysis might not be a bad idea, but probably better done in the html view.

**Pull request merged**